src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```
@@ -0,0 +1,74 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movie session " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getMovieSessionQuery = session.createQuery(
+                    "from MovieSession ms "
+                            + "left join fetch ms.movie "
+                            + "left join fetch ms.cinemaHall "
+                            + "where ms.id = :id", MovieSession.class
+            );
+            getMovieSessionQuery.setParameter("id", id);
+            return Optional.ofNullable(getMovieSessionQuery.getSingleResult());
```

This method uses `getSingleResult()` and then wraps it in `Optional.ofNullable(...)`. `getSingleResult()` throws NoResultException when there is no matching row, so the DAO will throw (and be wrapped into DataProcessingException) instead of returning Optional.empty(). This violates the checklist guidance: "* Think what the difference between `getSingleResult()`, `uniqueResult()` and `uniqueResultOptional()`? Which one should we use?" Replace this with a method that returns an Optional (e.g., `uniqueResultOptional()` or `getResultStream().findFirst()`) so DAO.get(...) returns Optional.empty() when not found.